### PR TITLE
Epic 7.1: Multiple Choice Generator - Virtual Vanguard

### PR DESCRIPTION
**Pull request by oddblaster**
Implemented support for more file types for the quizzify feature. Feature now supports all of the following file types: "xlsx", "pdf", "pptx", "csv", "docx","jpeg",'jpg',"png", "ppt", "html", youtube links, and google drive links.
Implemented robust support for more file types for the quizzify feature. Feature now supports all of the following file types: "xlsx", "pdf", "pptx", "csv", "docx","jpeg",'jpg',"png", "ppt", "html", youtube videos, and google drive links.

Multiple functionalities can be successfully uploaded from the backend.

Issues persists with attempting to upload multiple files.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (mikhailocampo) [inline .vscode/launch.json:1]: remove .vscode and related, place in .gitignore

File: app/features/Kaichat/metadata.json
Comment (mikhailocampo) [inline app/features/Kaichat/metadata.json:1]: i think we still need this?

File: app/features/quizzify/tools.py [hunk 2]
```diff
@@ -54,27 +85,77 @@ def chained_func(*args, **kwargs):
...
-                    doc = Document(page_content=page_content, metadata=metadata)
-                    documents.append(doc)
 
+            except Exception as e:
+                print(f"Error loading video {video_id}: {e}")
+       
         return documents
 
-class BytesFilePDFLoader:
+# class YoutubeLoaders:
```
Comment (mikhailocampo) [inline app/features/quizzify/tools.py:130]: Should remove commented out features that are not in use

File: app/features/quizzify/tools.py [hunk 3]
```diff
@@ -113,79 +376,129 @@ def load(self) -> List[Document]:
...
-                for i, page in enumerate(pdf_reader.pages):
-                    page_content = page.extract_text()
-                    metadata = {"source": file_path, "page_number": i + 1}
+                loader = self.loader_dict[file_type]
+                documents.extend(loader([file]).load())
+        return documents
+    
+class URLLoader():
+    def __init__(self, verbose=False):
+        self.expected_file_types = ["xlsx", "pdf", "pptx", "csv", "docx","jpeg",'jpg',"png", "ppt", "html",]
```
Comment (mikhailocampo) [inline app/features/quizzify/tools.py:390]: use enums for accepted types

**Review by mikhailocampo — COMMENTED** [review]
See my reviews

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    // Use IntelliSense to learn about possible attributes.
```
Comment (hash2004) [inline .vscode/launch.json:2]: Please remove unnecessary comments.

File: app/features/Kaichat/metadata.json
Comment (hash2004) [inline app/features/Kaichat/metadata.json:1]: This file should not be deleted.

File: app/features/quizzify/tools.py [hunk 5]
```diff
@@ -83,26 +126,200 @@ def load(self) -> List[Document]:
...
+        return documents
+    
+class ImageLoader(BaseLoader):
+    def __init__(self,files: List[Tuple[BytesIO,str]]):
+        self.files = files
+    
+    def load(self) -> List[Document]:
+        documents = []
+        text_completion_model = VertexAI(model='gemini-1.5-flash-001')
+        prompt= PromptTemplate.from_template("I want you to check if there is any missing words in {text}. If there are any, I want to to autocomplete them with the most relevant word possible and make the whole thing grammatically correct. The output should be a string.")
```
Comment (hash2004) [inline app/features/quizzify/tools.py:203]: There are better prompting techniques. You can gain inspiration from:https://www.promptingguide.ai/techniques

Comment (hash2004) [inline app/features/quizzify/tools.py:203]: Also, prompts should be stored in txt files or constants.py file.

File: app/requirements.txt
Comment (hash2004) [inline app/requirements.txt:1]: Add fastapi[standard]

File: app/utils/auth.py [hunk 6]
```diff
@@ -15,10 +15,11 @@ def access_secret_file(secret_id, version_id="latest"):
...
 def key_check(api_key: str = Header(None)):
   
-  if os.environ['ENV_TYPE'] == "production":
+  if os.environ.get('ENV_TYPE') == "production":
     set_key = access_secret_file("backend-access")
   else:
     set_key = "dev"
   
   if api_key is None or api_key != set_key:
+    print(api_key)
```
Comment (hash2004) [inline app/utils/auth.py:24]: You should not print api keys. It is not considered best practice.

**Review by hash2004 — CHANGES_REQUESTED** [review]

**Comment by AaronSosaRamos** [comment]
Closed due inactivity

**Pull request closed without merging**